packages/features/apps/AdminAppsList.tsx

```diff
@@ -0,0 +1,237 @@
+                    keys: values,
+                    dirName: app.dirName,
+                  })
+                }
+                className="px-4 pb-4">
+                {Object.keys(app.keys).map((key) => (
+                  <Controller
+                    name={key}
+                    key={key}
+                    control={formMethods.control}
+                    defaultValue={app.keys && app.keys[key] ? app?.keys[key] : ""}
+                    render={({ field: { value } }) => (
+                      <TextField
+                        label={key}
+                        key={key}
+                        name={key}
+                        value={value}
+                        onChange={(e) => {
+                          formMethods.setValue(key, e?.target.value);
+                        }}
+                      />
+                    )}
+                  />
+                ))}
+                <Button type="submit" loading={saveKeysMutation.isLoading}>
+                  {t("save")}
+                </Button>
+              </Form>
+            )}
+          </CollapsibleContent>
+        </div>
+      </Collapsible>
+
+      <Dialog open={disableDialog} onOpenChange={setDisableDialog}>
+        <ConfirmationDialogContent
+          title={t("disable_app")}
+          variety="danger"
+          onConfirm={() => {
+            enableAppMutation.mutate({ slug: app.slug, enabled: app.enabled });
+          }}>
+          {t("disable_app_description")}
+        </ConfirmationDialogContent>
+      </Dialog>
+    </>
+  );
+};
+
+const querySchema = z.object({
+  category: z
+    .nativeEnum({ ...AppCategories, conferencing: "conferencing" })
+    .optional()
+    .default(AppCategories.calendar),
+});
+
+const AdminAppsList = ({ baseURL, className }: { baseURL: string; className?: string }) => {
+  const { t } = useLocale();
+  const router = useRouter();
+  const { category } = querySchema.parse(router.query);
+
+  const { data: apps, isLoading } = trpc.viewer.appsRouter.listLocal.useQuery(
+    { category },
+    {
+      enabled: router.isReady,
+    }
+  );
+
+  return (
+    <AppCategoryNavigation
+      baseURL={baseURL}
+      containerClassname="w-full xl:mx-5 xl:w-2/3 xl:pr-5"
+      className={className}>
+      {(() => {
+        if (isLoading) return <SkeletonLoader />;
+        if (!apps) {
+          return (
+            <EmptyScreen
+              Icon={Icon.FiAlertCircle}
+              headline={t("no_available_apps")}
+              description={t("no_available_apps_description")}
+            />
+          );
+        }
+        return (
+          <div className="rounded-md border border-gray-200">
+            {apps.map((app, index) => (
+              <IntegrationContainer
+                app={app}
+                lastEntry={index === apps.length - 1}
+                key={app.name}
+                category={category}
+              />
+            ))}
+          </div>
+        );
+      })()}
```

Comment:
Since this is immediately invoked - why can't we just render this ?

Reply:
We can't assume that the data will be available immediately. Especially if we're aiming for 100 apps it might take a bit to load.